syntax = "proto3";
package pb;

//// 传入文件路径
//message AnalyseRequest {
//  string file_path = 1;
//}
message AnalyseRequest{
  bytes data = 1;
}
// 返回初步分析结果
message AnalyseResponse {
  int32 all_account = 1;         // 返回数据的所有值
  int32 level0 = 2;              // 提示数据的个数
  int32 level1 = 3;              // 次要数据的个数
  int32 level2 = 4;              // 主要数据的个数
  int32 level3 = 5;              // 紧急数据的个数
}
// 传入文件路径
message TrainRequest {
  bool active = 1;
}

// 返回训练完成标志
message TrainResponse {
  bool success = 1; //  是否完成训练
}


//传入的需要预测的数据结构
message Datainput{
   string clocationinfo = 1; // 设备定位信息
}

// 传入需要进行预测的告警序列
message AlertRequest {
  repeated Datainput datarequest = 1;  // 请求数据
}
// 前后交互的数据，传入参照
// 1028-9-上饶余干康山-OADM2:EMUP_F1K[00]:14
// 格式为int-int-string-string-string:string-string[]
message Dataoutput{
  string clocationinfo = 1;
}

// 预测结果
message AlertResponse {
  repeated Dataoutput dataresponse = 1;  // 返回的数据数组
}


// 定义服务接口
// 服务的顺序为：1.传入数据 2.进行训练 3.预测
service core {
  // 传入训练数据
  rpc Analyse (AnalyseRequest) returns (AnalyseResponse) {}
  // 训练
  rpc Train (TrainRequest) returns (TrainResponse) {}
  // 预测
  rpc Alert (AlertRequest) returns (AlertResponse) {}
}